solidity/test/rewards/TestECDSABackportRewards.js

```diff
@@ -0,0 +1,258 @@
+    )
+    const expectedLastSignerBalance =
+      expectedSingleRewardForLastSigner +
+      keepNumber * expectedSingleRewardForLastSigner
+
+    // Check the balance of the last beneficiary.
+    expect(actualLastSignerBalance).to.eq.BN(expectedLastSignerBalance)
+  }
+
+  async function initializeNewFactory() {
+    keepToken = await KeepToken.new({from: owner})
+    const keepTokenGrant = await KeepTokenGrant.new(keepToken.address)
+    registry = await KeepRegistry.new({from: owner})
+
+    bondedSortitionPoolFactory = await BondedSortitionPoolFactory.new({
+      from: owner,
+    })
+    await TokenStaking.detectNetwork()
+    await TokenStaking.link(
+      "MinimumStakeSchedule",
+      (await MinimumStakeSchedule.new({from: owner})).address
+    )
+    await TokenStaking.link(
+      "GrantStaking",
+      (await GrantStaking.new({from: owner})).address
+    )
+    await TokenStaking.link("Locks", (await Locks.new({from: owner})).address)
+    await TokenStaking.link("TopUps", (await TopUps.new({from: owner})).address)
+
+    const stakingEscrow = await TokenStakingEscrow.new(
+      keepToken.address,
+      keepTokenGrant.address,
+      {from: owner}
+    )
+
+    const stakeInitializationPeriod = 30 // In seconds
+
+    tokenStaking = await TokenStaking.new(
+      keepToken.address,
+      keepTokenGrant.address,
+      stakingEscrow.address,
+      registry.address,
+      stakeInitializationPeriod,
+      {from: owner}
+    )
+    tokenGrant = await TokenGrant.new(keepToken.address, {from: owner})
+
+    keepBonding = await KeepBonding.new(
+      registry.address,
+      tokenStaking.address,
+      tokenGrant.address,
+      {from: owner}
+    )
+    randomBeacon = await RandomBeaconStub.new({from: owner})
+    const bondedECDSAKeepMasterContract = await BondedECDSAKeep.new({
+      from: owner,
+    })
+    keepFactory = await BondedECDSAKeepFactoryStub.new(
+      bondedECDSAKeepMasterContract.address,
+      bondedSortitionPoolFactory.address,
+      tokenStaking.address,
+      keepBonding.address,
+      randomBeacon.address,
+      {from: owner}
+    )
+
+    await registry.approveOperatorContract(keepFactory.address, {from: owner})
+  }
+
+  async function createMembers() {
+    const membersArr = []
+
+    // 16 members in each keep
+    for (let i = 0; i < keepSize; i++) {
+      const operator = accounts[i]
+      const beneficiary = accounts[keepSize + i]
+      await tokenStaking.setBeneficiary(operator, beneficiary)
+      const member = {
+        operator: operator,
+        beneficiary: beneficiary,
+      }
+
+      membersArr.push(member)
+    }
+
+    return membersArr
+  }
+
+  async function createKeeps() {
+    const members = keepMembers.map((m) => m.operator)
+    for (let i = 0; i < numberOfCreatedKeeps; i++) {
+      await keepFactory.stubOpenKeep(
+        owner,
+        members,
+        tokenStaking.address,
+        keepFactory.address,
+        keepCreationTimestamp
```

Comment:
Can we create them at different timestamps? We don't need to equally distribute them between the interval but we can e.g. have them created over the course of 4 days, adding 2 hours to each timestamp in the loop.

Reply:
Done.